### PR TITLE
host/audio: move old LE Audio files to new directory

### PR DESCRIPTION
**Pull request by KKopyscinski**
LE Audio lives now in nimble/host/audio.

### CODE REVIEW

**Comment by KKopyscinski** [comment]
@MariuszSkamra PTAL

**Review by MariuszSkamra — CHANGES_REQUESTED** [review]

File: nimble/host/pkg.yml [hunk 1]
```diff
@@ -32,6 +32,9 @@ pkg.deps:
     - nimble
     - nimble/transport
 
+pkg.deps.BLE_ISO:
+    - nimble/host/audio
+
```
[the remark this answers is not in the document]
Comment (KKopyscinski) [reply nimble/host/pkg.yml:37]: maybe we should have one common config for audio? I can add one here

[the remark this answers is not in the document]
Comment (MariuszSkamra) [reply nimble/host/pkg.yml:37]: sounds good to me

File: nimble/host/audio/syscfg.yml [hunk 2]
```diff
@@ -17,5 +17,8 @@
 #
 
 syscfg.defs:
+    BLE_AUDIO:
+        description: 'Indicates that a BLE Audio is present.'
```
Comment (MariuszSkamra) [inline nimble/host/audio/syscfg.yml:21]: I'd move this config to `host/syscfg.yml`, just like `BLE_MESH` is defined.
Plus, I'd suggest to describe this option like:

```suggestion
        description: 'This option enables Bluetooth LE Audio support.'
```

Comment (KKopyscinski) [reply nimble/host/audio/syscfg.yml:21]: Fully agree :)

File: apps/btshell/pkg.yml [hunk 3]
```diff
@@ -35,6 +35,7 @@ pkg.deps:
     - nimble/host/services/gatt
     - nimble/host/store/config
     - nimble/host/util
+    - nimble/host/audio
```
[the remark this answers is not in the document]
Comment (KKopyscinski) [reply apps/btshell/pkg.yml:40]: moved to deps of BLE_AUDIO

**Review by MariuszSkamra — APPROVED** [review]
LGTM

File: nimble/host/syscfg.yml [hunk 4]
```diff
@@ -440,6 +440,11 @@ syscfg.defs:
             that have been enabled in the stack, such as GATT support.
         value: 0
 
+    BLE_AUDIO:
+        description: 'This option enables Bluetooth LE Audio support'
+        value: 0
```
Comment (sjanc) [inline nimble/host/syscfg.yml:445]: mark this as experimental

File: apps/btshell/src/cmd_leaudio.c [hunk 5]
```diff
@@ -17,13 +17,13 @@
...
 
-#include "host/ble_audio_broadcast_source.h"
 #include "cmd_leaudio.h"
 #include "btshell.h"
 #include "console/console.h"
 #include "errno.h"
 
-#if (MYNEWT_VAL(BLE_ISO_BROADCAST_SOURCE))
+#if (MYNEWT_VAL(BLE_AUDIO))
+#include "audio/ble_audio_bcst_source.h"
```
Comment (sjanc) [inline apps/btshell/src/cmd_leaudio.c:26]: I'm not very happy with this bcst abbreviation...

**Review by sjanc — COMMENTED** [review]
split this to separate commits ( CI, apps, host)

File: apps/btshell/src/cmd_leaudio.c (hunk 5 above)
Comment (KKopyscinski) [reply apps/btshell/src/cmd_leaudio.c:26]: So maybe ble_audio_broadcast_src? and later we'll use snk counterpart

Comment (KKopyscinski) [reply apps/btshell/src/cmd_leaudio.c]: Left full `broadcast` in names

File: nimble/host/syscfg.yml (hunk 4 above)
Comment (KKopyscinski) [reply nimble/host/syscfg.yml:445]: done

**Comment by KKopyscinski** [comment]
@sjanc as discussed offline, this refactor must be single-commit, because apps won't build otherwise

File: apps/auracast/syscfg.yml [hunk 6]
```diff
@@ -59,4 +59,6 @@ syscfg.vals:
     BLE_MAX_BIG: 1
     BLE_MAX_BIS: 2
 
-    BLE_PHY_NRF52_HEADERMASK_WORKAROUND: 1
\ No newline at end of file
+    BLE_PHY_NRF52_HEADERMASK_WORKAROUND: 1
```
Comment (andrzej-kaczmarek) [inline apps/auracast/syscfg.yml:62]: this is only required on nrf52 if encryption is enabled

File: nimble/host/syscfg.yml [hunk 7]
```diff
@@ -523,6 +514,20 @@ syscfg.defs:
         description: 'Minimum level for the BLE EATT log.'
         value: 1
 
+    BLE_MAX_BIG:
```
Comment (andrzej-kaczmarek) [inline nimble/host/syscfg.yml:517]: BLE_ISO_MAX_BIGS or BLE_AUDIO_MAX_BIGS

I think the former makes more sense

File: apps/btshell/pkg.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (andrzej-kaczmarek) [reply apps/btshell/pkg.yml:40]: dep should be added by package which defines BLE_AUDIO

File: nimble/host/syscfg.yml (hunk 7 above)
Comment (KKopyscinski) [reply nimble/host/syscfg.yml:517]: I left ISO in host, so I'll change to that

Comment (KKopyscinski) [reply nimble/host/syscfg.yml:517]: And BLE_ISO_MAX_BISES

File: apps/auracast/syscfg.yml (hunk 6 above)
Comment (KKopyscinski) [reply apps/auracast/syscfg.yml:62]: I'll remove it then

File: apps/btshell/pkg.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (KKopyscinski) [reply apps/btshell/pkg.yml]: done

**Review by andrzej-kaczmarek — APPROVED** [review]

**Review by sjanc — APPROVED** [review]

**Pull request merged**